qiskit_experiments/curve_analysis/curve_data.py

```diff
@@ -137,3 +139,205 @@ class ParameterRepr:
+        self,
+        parameters: List[str],
+        defaults: Optional[Union[Iterable[Any], Dict[str, Any]]] = None,
+    ):
+        """Create new dictionary.
+
+        Args:
+            parameters: List of parameter names used in the fit model.
+            defaults: Default values.
+
+        Raises:
+            AnalysisError: When defaults is provided as array-like but the number of
+                element doesn't match with the number of fit parameters.
+        """
+        if defaults is not None:
+            if not isinstance(defaults, dict):
+                if len(defaults) != len(parameters):
+                    raise AnalysisError(
+                        f"Default parameter {defaults} is provided with array-like "
+                        "but the number of element doesn't match. "
+                        f"This fit requires {len(parameters)} parameters."
+                    )
+                defaults = dict(zip(parameters, defaults))
+
+            full_options = {p: self.format(defaults.get(p, None)) for p in parameters}
+        else:
+            full_options = {p: None for p in parameters}
+
+        super().__init__(**full_options)
+
+    def __setitem__(self, key, value):
+        """Set value with validations.
+
+        Raises:
+            AnalysisError: When key is not previously defined.
+        """
+        if key not in self:
+            raise AnalysisError(f"Parameter {key} is not defined in this fit model.")
+
+        # value can be set if never assigned
+        super().__setitem__(key, self.format(value))
+
+    def __hash__(self):
+        return hash(tuple(sorted(self.items())))
+
+    def set_if_empty(self, **kwargs):
+        """Set value to the dictionary if not assigned.
+
+        Args:
+              kwargs: Key and new value to assign.
+        """
+        for key, value in kwargs.items():
+            if self.get(key) is None:
+                self.__setitem__(key, value)
+
+    @staticmethod
+    def format(value: Any) -> Any:
+        """Format dictionary value.
+
+        Args:
+            value: New value to assign.
+
+        Returns:
+            Formatted value.
+        """
+        return value
+
+
+class InitialGuesses(OptionsDict):
+    """Dictionary providing a validation for initial guesses."""
+
+    @staticmethod
+    def format(value: Any) -> Any:
+        """Validate if value is float.
+
+        Args:
+            value: New value to assign.
+
+        Returns:
+            Formatted value.
+
+        Raises:
+            AnalysisError: When value is invalid format.
+        """
+        if value is None:
+            return None
+
+        try:
+            return float(value)
+        except (TypeError, ValueError) as ex:
+            raise AnalysisError(f"Input value {value} is not valid initial guess. ") from ex
+
+
+class Boundaries(OptionsDict):
+    """Dictionary providing a validation for boundaries."""
+
+    @staticmethod
+    def format(value: Any) -> Any:
```

Comment:
How about? Lint seemed fine with this on my side.
```suggestion
    def format(value: Any) -> Optional[Tuple[float, float]]:
```